### ISSUE TITLE
Error with "import type { PageData } from './$types';" upgrading to v2

### ISSUE THREAD

**Issue by bluepuma77** [sveltejs/kit#11339]
### Describe the bug

Migrating SvelteKit app to v2. According to [docs](https://kit.svelte.dev/docs/routing#page-page-svelte) this is still valid:
```
<script lang="ts">
	import type { PageData } from './$types';
	export let data: PageData;
</script>
```
But it creates an error:
```
/home/projects/sveltejs-kit-template-default-vldzmz/src/routes/about/+page.svelte:2:13 Unexpected token

/home/projects/sveltejs-kit-template-default-vldzmz/src/routes/about/+page.svelte:2:13

1 |  <script lang="ts">
 2 |    import type { PageData } from './$types';
                    ^
 3 |    
 4 |    export let data: PageData;
```

### Reproduction

Create a new SvelteKit app with https://sveltekit.new/ and place this in a `+page.svelte`:
```
<script lang="ts">
	import type { PageData } from './$types';
	export let data: PageData;
</script>
```

### System Info

```Shell
Stackblitz latest
```

### Severity

blocking an upgrade

**Comment by bluepuma77** [comment]
I have the same problem locally. It worked before running `ncu -u && npm i` and later `npx svelte-migrate sveltekit-2`.

**Comment by dummdidumm** [comment]
That's just the JSDoc version, the code you have is still valid and nothing has changed on that end.

**Comment by bluepuma77** [comment]
All my fault. The different docs came from different browsers having different settings (JS/TS) for the docs.

TypeScript is not working anymore, because I removed the preprocessor. After SvelteKit 2 upgrade it threw some errors, either that or some doc I googled said to remove `vitepreprocess`. I didn't under stand it was removing Typescript support.

https://kit.svelte.dev/docs/integrations#preprocessors-vitepreprocess
